dll/win32/shell32/shlexec.cpp

```diff
@@ -624,16 +644,38 @@ static BOOL SHELL_TryAppPathW( LPCWSTR szName, LPWSTR lpResult, WCHAR **env)
         if (res) goto end;
     }
 
-    len = MAX_PATH * sizeof(WCHAR);
-    res = RegQueryValueW(hkApp, NULL, lpResult, &len);
-    if (res) goto end;
+    len = sizeof(szPath);
+    res = RegQueryValueExW(hkApp, NULL, NULL, &dwType, (LPBYTE)szPath, &len);
+    if (res || (dwType != REG_SZ && dwType != REG_EXPAND_SZ))
+        goto end;
+
+    if (dwType == REG_EXPAND_SZ)
+        ExpandEnvironmentStringsW(szPath, lpResult, MAX_PATH);
+    else
+        lstrcpynW(lpResult, szPath, MAX_PATH);
```

Comment:
In principle this whole job of query + expand may be done by `SHRegQueryValueExW `
but as we can see from here the expansion part seems to be unimplemented:
https://git.reactos.org/?p=reactos.git;a=blob;f=dll/win32/shell32/wine/shellreg.c;hb=c8ef276778854a1eec6bd6ec1f6377ef9b023189#l103

Reply:
OK.